dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/DefaultAppManager.java

```diff
@@ -405,7 +414,13 @@ public void reloadApps() {
         .filter(app -> !exists(app.getKey()))
         .forEach(this::installApp);
 
-    jCloudsAppStorageService.discoverInstalledApps().values().forEach(this::installApp);
+    jCloudsAppStorageService.discoverInstalledApps().values().stream()
```

Comment:
+1 for David's point - a comment here would be great (especially that order matters, I take it - this works because we start with jCloudApps before checking bundleAppStorage from what I can see)

in all cases, it took me a bit of time to figure it out